TESTS/mbed_drivers/flashiap/main.cpp

```diff
@@ -28,6 +28,30 @@
 
 using namespace utest::v1;
 
+// Find ROM end in flash, depending on the toolchain, so our test doesn't overwrite it
+#if defined(TOOLCHAIN_GCC_ARM)
+extern uint32_t __etext;
+#define ROM_END ((uint32_t) &__etext)
+#elif defined(TOOLCHAIN_ARM)
+extern uint32_t Load$$LR$$LR_m_text$$Limit[];
+#define ROM_END ((uint32_t)Load$$LR$$LR_m_text$$Limit)
+#elif defined(TOOLCHAIN_IAR)
+#pragma section=".text"
+extern void *rom_section_end  = __section_end(".text");
+#define ROM_END ((uint32_t)rom_section_end)
+#else
+#define ROM_END 0
+#error [NOT_SUPPORTED] Unsupported toolchain
+#endif
+
+// This macro returns if flash test overwrites the code in flash.
+#define RETURN_IF_TEST_OVERWRITES_CODE(ADDRESS) \
```

Comment:
this would be nicer if its just a function, no need for a macro?

Reply:
I'm not a big macro fan myself. However, in this case, macro seems like the better solution, as it includes a `return` inside. Turning this into a function would make the code a bit more bloated, as it would make all three tests check the return value, and return if it indicates that the code is getting overwritten. No biggie, can change if required.

Reply:
👍  Fine as it is